src/lowering/desugar.ml

```diff
@@ -464,6 +464,64 @@ and export_footprint self_id expr =
   )],
   [{ it = I.{ name = lab; var = v }; at = no_region; note = typ }])
 
+and export_runtime_information self_id =
+  let open T in
+  let {lab;typ;_} = motoko_runtime_information_fld in
+  let v = "$"^lab in
+  let rts_memory_size = fresh_var "memorySize" T.nat in
+  let rts_heap_size = fresh_var "heapSize" T.nat in
+  let rts_total_allocation = fresh_var "totalAllocation" T.nat in
+  let rts_reclaimed = fresh_var "reclaimed" T.nat in
+  let rts_max_live_size = fresh_var "maxLiveSize" T.nat in
+  let rts_stable_memory_size = fresh_var "stableMemorySize" T.nat in
+  let rts_logical_stable_memory_size = fresh_var "logicalStableMemorySize" T.nat in
+  let rts_max_stack_size = fresh_var "maxStackSize" T.nat in
+  let rts_callback_table_count = fresh_var "callbackTableCount" T.nat in
+  let rts_callback_table_size = fresh_var "callbackTableSize" T.nat in
```

Comment:
NIT: Since they all have the same type maybe once could shorten this using some list functions to construct a list of declarations and list of field types and then use them below....

Reply:
Thanks. I refactored it.